File: 백준/Bronze/4153. 직각삼각형/직각삼각형.cc
#include <stdio.h>
#include <iostream>
#include <vector>
#include <cmath>
#include <algorithm>
#include <set>
#include <ostream>
#include <string>
#include <cstring>
#include <sstream>
#include <time.h>
#include <queue>
#include <cstdlib>
#include <stdbool.h>
#include <map>

#define fastio ios::sync_with_stdio(0), cin.tie(0), cout.tie(0)
using namespace std;

int x, y, z;
int arr[3];

int main(){
    fastio;

    while(true){
        cin >> x >> y >> z;
        
        if (x == 0 && y == 0 && z == 0) break;
    
        arr[0] = x; arr[1] = y; arr[2] = z;
        
        sort(arr, arr+3);
        
        if ((arr[0] * arr[0]) + (arr[1] * arr[1]) == (arr[2] * arr[2])) cout << "right" << '\n';
        else cout << "wrong" << '\n';
    }
    
    
    
    return 0;
}

File: 백준/Bronze/4153. 직각삼각형/직각삼각형.java
import java.util.Arrays;
import java.util.Scanner;

public class Main {
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		while (true) {
			int a = sc.nextInt(), b = sc.nextInt(), c = sc.nextInt();
			if (a == 0 && b == 0 && c == 0) {
				break;
			}
			int[] arr = {a, b, c};
			Arrays.sort(arr);
			if (arr[0] * arr[0] + arr[1] * arr[1] == arr[2] * arr[2]) {
				System.out.println("right");
			} else {
				System.out.println("wrong");
			}
		}
	}
}